server/routes/blackjack.py

```diff
@@ -74,3 +123,112 @@ def bet_blackjack():
         }
     except json.decoder.JSONDecodeError:
         return {"error": "Malformed request"}, 400
+
+@blackjack_bp.route("/api/blackjack/playagain", methods=["POST"])
+def play_again_blackjack():
+    try:
+        data = json.loads(request.data)
+        blackjack_transaction(-data["amount"])
+        new_cards = get_starting_cards()
+        client_dealer = translate_hand(new_cards["dealer_hand"])[0:2]
+        client_player = translate_hand(new_cards["player_hand"])
+        return {
+            "success": True,
+            "blackjack" : blackjack_total(new_cards["player_hand"]) == 21,
+            "dealer": client_dealer,
+            "player": client_player,
+        }
+    except json.decoder.JSONDecodeError:
+        return {"error": "Malformed request"}, 400
+
+@blackjack_bp.route("/api/blackjack/hit", methods=["GET"])
+def hit_blackjack():
+    query = Blackjack.query.filter_by(user_id=session["user_id"]).first()
+    deck = json.loads(query.deck)
+    player_hand = json.loads(query.player_hand)
+    next_card = draw_card(deck)
+    if len(deck)<200:
+        deck = deck + get_deck_set()
+    player_hand.append(next_card)
+    query.deck = json.dumps(deck)
+    query.player_hand = json.dumps(player_hand)
+    db.session.commit()
+    total = blackjack_total(player_hand)
+    client_player = translate_hand(player_hand)
+    if total > 21:
+        return {
+            "success": True,
+            "bust": True,
+            "winner": "Dealer",
+            "blackjack": False,
+            "player": client_player
+        }
+    if total == 21:
+        return {
+            "success": True,
+            "bust": False,
+            "blackjack" : True,
+            "player": client_player,
+        }
+    return {
+        "success": True,
+        "bust": False,
+        "blackjack": False,
+        "player": client_player
+        }
+
+@blackjack_bp.route("/api/blackjack/stand", methods=["GET"])
+def stand_blackjack():
+    query = Blackjack.query.filter_by(user_id=session["user_id"]).first()
+    deck = json.loads(query.deck)
+    player_hand = json.loads(query.player_hand)
+    dealer_hand = json.loads(query.dealer_hand)
+    while blackjack_total(dealer_hand) <= 17:
+        dealer_hand.append(draw_card(deck))
+    if len(deck)<200:
+        deck = deck + get_deck_set()
+    query.deck = json.dumps(deck)
+    query.player_hand = json.dumps(player_hand)
+    query.dealer_hand = json.dumps(dealer_hand)
+
+    client_dealer = translate_hand(dealer_hand)
+    client_player = translate_hand(player_hand)
+    dealer_total = blackjack_total(dealer_hand)
+    player_total = blackjack_total(player_hand)
+    if dealer_total > 21 or dealer_total < player_total:
+        last_transaction = db.session.query(Transaction)\
+            .filter_by(user_id=session["user_id"])\
+            .order_by(Transaction.id.desc()).first()
+        new_amount = last_transaction.ticket_amount * 1.5
```

Comment:
This is exploitable. A player could bet a small amount and if he knows he's about to win, he could bet a lot of tokens on coinflip. He can continue until he loses. Then, he'll get 1.5 times the last transaction amount.
For example, I bet 500 tickets on blackjack, when I get 21 and I'm likely to win, I bet 1,000,000 on coinflip. If I win, I continue until I lose. When blackjack is over, I'll get 1,500,000 tickets after betting just 500 tickets.

Also, why is the multiplier 1.5? Shouldn't it be 2? You earn the amount of money you bet (1) and the money you bet (1) = 1+1 = 2.

Reply:
Looked this up. Player blackjack is 1.5 but player vs house is 2.

Reply:
2 makes more sense here since this is a dealer blackjack.

Reply:
I fixed it so that it also searches only for blackjack as activity in case someone tries that. The new gain is *2.

Reply:
Please check if that is good or there is another hole.